# SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note
diff --git a/drivers/hwmon/pmbus/tps544.c b/drivers/hwmon/pmbus/tps544.c
index b91c77301369..22540d957a6b 100644
--- a/drivers/hwmon/pmbus/tps544.c
+++ b/drivers/hwmon/pmbus/tps544.c
@@ -359,3 +359,4 @@ module_i2c_driver(tps544_driver);
 MODULE_AUTHOR("Harini Katakam");
 MODULE_DESCRIPTION("PMBus regulator driver for TPS544");
 MODULE_LICENSE("GPL");
+MODULE_IMPORT_NS(PMBUS);
